### PR TITLE
Finalize bitbucket api

### PR DESCRIPTION
**Pull request by ManuelFeller**
- finalized Bitbucket API
  - auto branch creation
  - auto file creation
- added some tests for internal classes without external dependencies

Manually "integration-tested" on Directus 9.5.1 against Bitbucket Server 6.10.12 without any errors

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+	"cSpell.words": [
+		"Directus"
+	]
+}
```
Comment (gerardcl) [inline .vscode/settings.json]: new line missing EOF

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  preset: 'ts-jest',
+  testEnvironment: 'node',
+};
```
Comment (gerardcl) [inline jest.config.js]: new line missing EOF

File: src/api/BitbucketVersionControl.ts [hunk 3]
```diff
@@ -61,52 +61,234 @@ export default class BitbucketVersionControl implements VersionControlAbstractio
...
+		} catch (error: any) {
+			this.loggerReference.logObject(LogLevels.debug, 'Axios error result:', error.response.data.errors);
+		}
+		if (response.status !== 200) {
+			throw this.exceptionProvider.getNewException(ExceptionTypes.UnexpectedResponseException,
+					'BitbucketVersionControl: could not retrieve list of branches');
+		}
+		if (!response.data.isLastPage) {
+			throw this.exceptionProvider.getNewException(ExceptionTypes.UnexpectedResponseException,
+					'BitbucketVersionControl: found more then 1000 branches in the repository, canceling changelog commit - please clean up to reactivate changelog writing');
```
Comment (gerardcl) [inline src/api/BitbucketVersionControl.ts]: could make 1000 a constant

File: src/api/BitbucketVersionControl.ts [hunk 4]
```diff
@@ -61,52 +61,234 @@ export default class BitbucketVersionControl implements VersionControlAbstractio
...
+	 */
+	private async createBranch(branchName: string, startCommit: string): Promise<void> {
+		const bitbucketApiAuthHeader = {
+			auth: {
+				username: this.authentication.user,
+				password: this.authentication.password
+			}
+		};
+		this.loggerReference.logMessage(LogLevels.debug, 'BitbucketVersionControl: creating working branch in repository');
+		let responseStatus = undefined;
```
Comment (gerardcl) [inline src/api/BitbucketVersionControl.ts]: in previous method this was given type `any`

File: src/api/BitbucketVersionControl.ts [hunk 5]
```diff
@@ -61,52 +61,234 @@ export default class BitbucketVersionControl implements VersionControlAbstractio
...
+		return response.data.values[0].id as string;
+	}
 
+	/**
+	 * Internal function to get the content of the current changelog in the repository
+	 * @returns An array os strings, each item represents a line in the file
+	 */
+	private async getCurrentChangelogContent(): Promise<Array<string>> {
+		let result = new Array<string>();
+		const bitbucketApiAuthHeader = {
```
Comment (gerardcl) [inline src/api/BitbucketVersionControl.ts]: looks likethis auth header could be reused

**Review by gerardcl — APPROVED** [review]
really cool! 🚀 

I just added some comments to take into account but approving! 👍

File: src/api/BitbucketVersionControl.ts (hunk 4 above)
Comment (ManuelFeller) [reply src/api/BitbucketVersionControl.ts]: good find, changed

File: src/api/BitbucketVersionControl.ts (hunk 5 above)
Comment (ManuelFeller) [reply src/api/BitbucketVersionControl.ts]: makes sense, implemented

File: src/api/BitbucketVersionControl.ts (hunk 3 above)
Comment (ManuelFeller) [reply src/api/BitbucketVersionControl.ts]: also good find, overlooked that

File: jest.config.js (hunk 2 above)
Comment (ManuelFeller) [reply jest.config.js]: added

File: .vscode/settings.json (hunk 1 above)
Comment (ManuelFeller) [reply .vscode/settings.json]: complain to the VS Code people, that file is auto-generated :D - but also done

**Pull request merged**